Week 2/Weekly Practice Problems (Week 2)/2553: Separate the Digits in an Array/delight2553.java

```diff
@@ -0,0 +1,43 @@
+/*Solution 1 using String*/
+class Solution {
+    public int[] separateDigits(int[] nums) {
+        String temp;
+        ArrayList<Integer> list = new ArrayList<>();
+        for(int i = 0; i < nums.length; i++){
+            temp = Integer.toString(nums[i]);
+            for(int j = 0; j < temp.length(); j++){
+                list.add(temp.charAt(j)-'0');
+            }    
+        }
+        int[] returnArray = new int[list.size()];       
+        for (int i = 0; i < list.size(); i++) {       
+            returnArray[i] = list.get(i);                
+        }                                        
+        return returnArray;
+    } 
+}
+/*Solution 2 using int*/
```

Comment:
Int is usually preferred because it's more light weight.

But the difference is not huge :)

Reply:
How do you know a code is more light weight? Lightweight means less memory, less dependencies, and faster computing time right? Both low time and space complexity?